src/hatch/cli/__init__.py

```diff
@@ -90,9 +90,26 @@
     envvar=ConfigEnvVars.CONFIG,
     help="The path to a custom config file to use [env var: `HATCH_CONFIG`]",
 )
+@click.option(
+    "--keep-env",
```

Comment:
This seems out of place. Can we make this an environment variable instead, or support an environment variable here and hide the option?